week1/search.py

```diff
@@ -111,11 +111,89 @@ def create_query(user_query, filters, sort="_score", sortDir="desc"):
     query_obj = {
         'size': 10,
         "query": {
-            "match_all": {} # Replace me with a query that both searches and filters
+            "function_score": {
+                "query": {
+                    "bool": {
+                        "must": [
+                            {
+                                "query_string": {
+                                    "fields": [
+                                        "name^100", 
+                                        "shortDescription^50",
+                                        "longDescription^10"],
+                                    "query": user_query,
+                                    "phrase_slop": 3,
+                                }
+                            }
+                        ],
+                        "filter": filters
+                    }
+                },
+                "boost_mode": "replace",
+                "score_mode": "avg",
+                "functions": [
+                    {
+                        "field_value_factor": {
+                            "field": "salesRankShortTerm",
+                            "modifier": "reciprocal",
+                            "missing": 100000000
+                        }
+                    },
+                    {
+                        "field_value_factor": {
+                            "field": "salesRankMediumTerm",
+                            "modifier": "reciprocal",
+                            "missing": 100000000
+                        }
+                    },
+                    {
+                        "field_value_factor": {
+                            "field": "salesRankLongTerm",
+                            "modifier": "reciprocal",
+                            "missing": 100000000
+                        }
+                    }
+                ]
+            }
         },
         "aggs": {
             #### Step 4.b.i: create the appropriate query and aggregations here
-
-        }
+            "regularPrice": {
+                "range": {
+                    "field": "regularPrice",
+                    "ranges": [
+                        { "key": "$", "to": 24.99 },
+                        { "key": "$$", "from": 25, "to": 49.99 },
+                        { "key": "$$$", "from": 50, "to": 74.99 },
+                        { "key": "$$$$", "from": 75, "to": 99.99 },
+                        { "key": "$$$$$", "from": 100, "to": 199.99 },
+                        { "key": "$$$$$$", "from": 200 }
+                    ]
+                }
+            },
+            "department": {
+                "terms": {
+                    "field": "department",
+                    "size": 10,
+                    "min_doc_count": 0
+                }
+            },
+            "missing_images": {
+                "missing": { "field": "image" }
+            }
+        },
+        "highlight": {
+            "pre_tags" : ["<strong>"],
```

Comment:
nice! I left the <em> default.

Reply:
thanks! was not trully catching the defaults (`<em>`) on the UI and this allowed me to properly see them in action